### PR TITLE
Refactor to use ID generator service for certificate ID

### PR DESCRIPTION
**Pull request by ethatcher83**
Remove sequence generator 
Add ID generator service 
Update tests

Resolves: GCI-1100

### CODE REVIEW

File: src/test/java/uk/gov/companieshouse/certificates/orders/api/controller/CertificateItemsControllerIntegrationTest.java [hunk 1]
```diff
@@ -65,12 +93,15 @@
...
+    private IdGeneratorService idGeneratorService;
+
     @MockBean
     private CompanyService companyService;
 
     private static final String CERTIFICATES_URL = "/orderable/certificates/";
-    private static final String EXPECTED_ITEM_ID = "CHS00000000000000001";
-    private static final String UPDATED_ITEM_ID  = "CHS00000000000000002";
+    private static final String EXPECTED_ITEM_ID = "CRT-123456-123456";
+    private static final String UPDATED_ITEM_ID  = "ORD-123456-123457";
```
Comment (VaughanJackson123) [inline src/test/java/uk/gov/companieshouse/certificates/orders/api/controller/CertificateItemsControllerIntegrationTest.java:104]: Probably should make this `CRT-...` too.

File: src/test/java/uk/gov/companieshouse/certificates/orders/api/controller/CertificateItemsControllerIntegrationTest.java [hunk 2]
```diff
@@ -273,7 +304,6 @@ void createCertificateItemSuccessfullyCreatesCertificateItem() throws Exception
         newItem.setCustomerReference(CUSTOMER_REFERENCE);
 
         final CertificateItemDTO expectedItem = new CertificateItemDTO();
-        expectedItem.setId(EXPECTED_ITEM_ID);
```
Comment (nabeelamirch) [inline src/test/java/uk/gov/companieshouse/certificates/orders/api/controller/CertificateItemsControllerIntegrationTest.java:276]: We should probably still assert that the value of the id generated from the service, is in fact the id in the response for this test.

File: src/test/java/uk/gov/companieshouse/certificates/orders/api/controller/CertificateItemsControllerIntegrationTest.java (hunk 1 above)
Comment (ethatcher83) [reply src/test/java/uk/gov/companieshouse/certificates/orders/api/controller/CertificateItemsControllerIntegrationTest.java:104]: Thanks, updated

File: src/test/java/uk/gov/companieshouse/certificates/orders/api/controller/CertificateItemsControllerIntegrationTest.java (hunk 2 above)
Comment (ethatcher83) [reply src/test/java/uk/gov/companieshouse/certificates/orders/api/controller/CertificateItemsControllerIntegrationTest.java:276]: Thanks, added back in

**Pull request merged**